ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/CloudSourceIntegrations/OcmIntegrationForm.tsx

```diff
@@ -0,0 +1,419 @@
+            secret: '',
+            clientId: '',
+            clientSecret: '',
+        },
+        skipTestIntegration: true,
+        ocm: {
+            endpoint: 'https://api.openshift.com',
+        },
+    },
+    updateCredentials: true,
+};
+
+function OcmIntegrationForm({
+    initialValues = null,
+    isEditable = false,
+}: IntegrationFormProps<CloudSourceIntegration>): ReactElement {
+    const formInitialValues = structuredClone(defaultValues);
+    if (initialValues) {
+        formInitialValues.cloudSource = merge({}, formInitialValues.cloudSource, initialValues);
+        formInitialValues.cloudSource.credentials.secret = '';
+        formInitialValues.cloudSource.credentials.clientId = '';
+        formInitialValues.cloudSource.credentials.clientSecret = '';
+        formInitialValues.updateCredentials = false;
+    }
+    const {
+        values,
+        touched,
+        errors,
+        dirty,
+        isValid,
+        setFieldValue,
+        handleBlur,
+        isSubmitting,
+        isTesting,
+        onSave,
+        onTest,
+        onCancel,
+        message,
+    } = useIntegrationForm<CloudSourceIntegrationFormValues>({
+        initialValues: formInitialValues,
+        validationSchema,
+    });
+
+    const { isCreating } = usePageState();
+
+    function onChange(value, event) {
+        return setFieldValue(event.target.id, value);
+    }
+
+    function onUpdateCredentialsChange(value, event) {
+        setFieldValue('cloudSource.credentials.secret', '');
+        setFieldValue('cloudSource.credentials.clientId', '');
+        setFieldValue('cloudSource.credentials.clientSecret', '');
+        return setFieldValue(event.target.id, value);
+    }
+
+    return (
+        <>
+            <PageSection variant="light" isFilled hasOverflowScroll>
+                <FormMessage message={message} />
+                <Form isWidthLimited>
+                    <FormLabelGroup
+                        isRequired
+                        label="Integration name"
+                        fieldId="cloudSource.name"
+                        touched={touched}
+                        errors={errors}
+                    >
+                        <TextInput
+                            isRequired
+                            type="text"
+                            id="cloudSource.name"
+                            value={values.cloudSource.name}
+                            onChange={(event, value) => onChange(value, event)}
+                            onBlur={handleBlur}
+                            isDisabled={!isEditable}
+                        />
+                    </FormLabelGroup>
+                    <FormLabelGroup
+                        isRequired
+                        label="Endpoint"
+                        labelIcon={
+                            <IntegrationHelpIcon
+                                helpTitle="OpenShift Cluster Manager endpoint"
+                                helpText={
```

Comment:
Off the main subject of this review:
* On one side of the coin, `Flex` and `FlexItem` pattern is forward thinking in case of need to change later.
* On other side of the coin, if you feel like helpText="The API endpoint and so on…" is a useful alternative, we can adjust type of `IntegrationHelpIcon` component to allow it in an independent follow up.